test/transform/forEachTest.js

```diff
@@ -0,0 +1,328 @@
+        '    sell(fruit, price);\n' +
+        '  }\n' +
+        '}'
+      ).toReturn(
+        'fruitList.forEach(fruit => {\n' +
+        '  let price = getCurrentPrice(fruit);\n' +
+        '  if (price > fruit.standardPrice) {\n' +
+        '    sell(fruit, price);\n' +
+        '  }\n' +
+        '});'
+      );
+    });
+
+    it('should support complex expressions for array', () => {
+      expectTransform(
+        'for (let i=0; i < store[current].fruits.length; i++) {\n' +
+        '  let item = store[current].fruits[i];\n' +
+        '  console.log(item);\n' +
+        '}'
+      ).toReturn(
+        'store[current].fruits.forEach(item => {\n' +
+        '  console.log(item);\n' +
+        '});'
+      );
+    });
+
+    it('should preserve original const-kind of loop variable', () => {
+      expectTransform(
+        'for (let i=0; i < array.length; i++) {\n' +
+        '  const item = array[i];\n' +
+        '  console.log(item);\n' +
+        '}'
+      ).toReturn(
+        'array.forEach(item => {\n' +
+        '  console.log(item);\n' +
+        '});'
+      );
+    });
+
+    it('should transform ++i loop-increment', () => {
+      expectTransform(
+        'for (let i=0; i < array.length; ++i) {\n' +
+        '  let item = array[i];\n' +
+        '  console.log(item);\n' +
+        '}'
+      ).toReturn(
+        'array.forEach(item => {\n' +
+        '  console.log(item);\n' +
+        '});'
+      );
+    });
+
+    it('should transform i+=1 loop-increment', () => {
+      expectTransform(
+        'for (let i=0; i < array.length; i+=1) {\n' +
+        '  let item = array[i];\n' +
+        '  console.log(item);\n' +
+        '}'
+      ).toReturn(
+        'array.forEach(item => {\n' +
+        '  console.log(item);\n' +
+        '});'
+      );
+    });
+
+    it('should transform when index identifier used as object literal key', () => {
+      expectTransform(
+        'for (let i=0; i < array.length; i++) {\n' +
+        '  let item = array[i];\n' +
+        '  console.log(item, {i: 123});\n' +
+        '}'
+      ).toReturn(
+        'array.forEach(item => {\n' +
+        '  console.log(item, {i: 123});\n' +
+        '});'
+      );
+    });
+
+    it('should transform when index identifier used as object property', () => {
+      expectTransform(
+        'for (let i=0; i < array.length; i++) {\n' +
+        '  let item = array[i];\n' +
+        '  console.log(item.i);\n' +
+        '}'
+      ).toReturn(
+        'array.forEach(item => {\n' +
+        '  console.log(item.i);\n' +
+        '});'
+      );
+    });
+
+    it('should transform when array is used in loop body', () => {
+      expectTransform(
+        'for (let i=0; i < array.length; i++) {\n' +
+        '  let item = array[i];\n' +
+        '  console.log(array);\n' +
+        '}'
+      ).toReturn(
+        'array.forEach((item, i, array) => {\n' +
+        '  console.log(array);\n' +
+        '});'
+      );
+    });
```

Comment:
Do we really want to add the third array parameter? The code will work just fine without it.

Personally I don't remember a single time I've needed the third parameter of `forEach` function. I'd guess one might need it when the function is defined elsewhere and can't therefore access the whole array - but in the case of function literal right inside `forEach` there is no such problem.